lib/hls/hls_parser.js

```diff
@@ -2829,10 +2832,16 @@ shaka.hls.HlsParser = class {
       }
     }
 
+    const getUris = () => {
+      const absoluteSegmentUri = this.variableSubstitution_(
```

Comment:
Does `variableSubstitution_` take a significant amount of time?

Reply:
It depends on the device, on PC or Mac there are no problem, but on SmartTVs yes. Anyway this is the first change here, in another PR I will change how `hlsSegment.absoluteUri` is used to improve performance.